### ISSUE TITLE
Support h2 databases

### ISSUE THREAD

**Issue by the456gamer** [beekeeper-studio/beekeeper-studio#298]
Not sure if this is possible/feasable, but h2 support would be useful

https://www.h2database.com/html/main.html

**Comment by rathboma** [comment]
Agreed! This will require the JDBC connector, which I haven't tested yet, but is certainly possible.

**Comment by not-night-but** [comment]
@andersonbosa Contributions are always welcome! I'm not really sure what this would require (specifically with the JDBC connector), but the areas of code you should be looking at are the clients in `apps/studio/src/lib/db/clients/` directory. Every client we have is a child of the `BasicDatabaseClient` class (`apps/studio/src/lib/db/clients/BasicDatabaseClient.ts`). I'd look at the client for a db you're familiar with to figure out what to do. A connection form will also be needed (`apps/studio/src/components/connection/`). There will also need to be a mapping added in `apps/studio/src-commercial/backend/lib/db/client.ts`. There are probably other places you'll need to add stuff but the compiler will probably help with that. You can also message me on slack if you want help
